clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -110,7 +110,13 @@ public ConsumerCoordinator(ConsumerNetworkClient client,
 
         addMetadataListener();
 
-        this.autoCommitTask = autoCommitEnabled ? new AutoCommitTask(autoCommitIntervalMs) : null;
+        if (autoCommitEnabled) {
```

Comment:
That might make sense. There is some risk of surprising behavior with this patch for users who are using manual assignment with default configuration. In that case, we should probably change the behavior of commitSync() and commitAsync() to raise an exception if the groupId is empty.

I discussed this problem some time ago with @guozhangwang. I think we have to allow commits on the server with an empty groupId for compatibility with the old consumer, but we do at least require that the groupId is non-empty when joining the group. That leaves only offset commits which are not part of a joined group and maybe it's ok for the new consumer to not allow them.